lib/types/src/memory.rs

```diff
@@ -113,10 +121,89 @@ unsafe impl MemorySize for Memory64 {
     type Offset = u64;
     type Native = i64;
     const ZERO: Self::Offset = 0;
+    const ONE: Self::Offset = 1;
     fn offset_to_native(offset: Self::Offset) -> Self::Native {
         offset as Self::Native
     }
     fn native_to_offset(native: Self::Native) -> Self::Offset {
         native as Self::Offset
     }
 }
+
+/// Represents memory that is used by the WebAsssembly module
+pub trait LinearMemory
```

Comment:
Can we move this, along with the `VMMemoryDefinition` back to the `wasmer-vm` crate?
The JS implementation will not make use of it (and can't make use of it), so the VM seems a much better place